### DISCUSSION TITLE
useQuery + Server Actions = Refetch doesn't work.

### DISCUSSION THREAD

**Discussion by mailok** [vercel/next.js#65915] [App Router]
Hi, I need some help related to Server Actions. I'm using a Server Action with the react-query hook useQuery, and it only works the first time. After that, it completely stops working because refetching doesn't work. When I execute refetch, the server action runs, but the query stops functioning entirely. It never goes back to isLoading or updates with new data. It's really strange, and I have an example where this can be reproduced.

Does anyone have any ideas, suggestions, or anything?

https://github.com/mailok/query-server-actions

**Comment by carstenblt** [comment]
I am also running into this. Did you solve it?
